File: app/src/main/java/leavesc/hello/network/http/basis/callback/RequestCallback.java
package leavesc.hello.network.http.basis.callback;

/**
 * 作者：leavesC
 * 时间：2018/10/27 20:53
 * 描述：
 * GitHub：https://github.com/leavesC
 */
public interface RequestCallback<T> {

    void onSuccess(T t);

}

File: app/src/main/java/leavesc/hello/weather/core/http/RequestCallback.kt
package leavesc.hello.weather.core.http

/**
 * 作者：leavesC
 * 时间：2019/5/31 10:47
 * 描述：
 */
interface RequestCallback<T> {

    fun onSuccess(data: T)

}

interface RequestMultiplyCallback<T> : RequestCallback<T> {

    fun onFail(exception: BaseException)

}

interface RequestMultiplyToastCallback<T> : RequestMultiplyCallback<T>